hail/python/hail/ggplot/geoms.py

```diff
@@ -44,7 +44,7 @@ def _update_legend_trace_args(self, trace_args, legend_cache):
                 trace_args["showlegend"] = False
             else:
                 trace_args["showlegend"] = True
-                legend_cache.add(trace_args["name"])
+                legend_cache[trace_args["name"]] = True
```

Comment:
On 197, you `**prev` where prev is a value from the legend_cache. `**True` will error. Should this be assigning `{}` instead?

Reply:
this doesn't cause problems on 197 because `GeomPoint.apply_to_fig` never calls `_update_legend_trace_args`. i changed `legend_cache` to a `dict` instead of a `set` because i needed to put the whole `legend_group` in there rather than just the `aes_name`, but since all the other geoms still function the same as before, this was just the easiest way i could think of to have it basically still be a `set` (just the set of the `dict`'s keys, while ignoring the values)

that said, i think it's still a good idea to update it, so that the type of `legend_cache` is more consistent!